sessionctx/variable/tidb_vars.go

```diff
@@ -609,7 +609,7 @@ const (
 	DefSkipASCIICheck                  = false
 	DefOptAggPushDown                  = false
 	DefOptBCJ                          = false
-	DefOptCartesianBCJ                 = 1
+	DefOptCartesianBCJ                 = 2
```

Comment:
why change the default value? When the value is 2, it means for carstesian join, it will always generate a broadcast join plan even if the table size of both join side are very big, it will definitely cause some other problems(like OOM). I think it does not make sense to use 2 as the default value, in fact, I think this variabe should never be set to 2 in a serious environment, it should only be used when user tries to make some manual experiments. And even if the experiment result shows broadcast cartesian join is fast, user should adjust the threshold of broadcast join to let optimizer choose broadcast cartesian join.

Reply:
after discussed, we found that this join will cause oom in both TiDB and TiFlash, and the TiDB's memory control strategy is better than tiflash, so we keep its default 1 lest import extra risks.